kotlin/tests/src/test/kotlin/com/squareup/moshi/kotlin/DualKotlinTest.kt

```diff
@@ -436,6 +436,35 @@ class DualKotlinTest(useReflection: Boolean) {
       @Suppress("REDUNDANT_PROJECTION")
       val complex: TypeAliasGeneric<List<out TypeAliasGeneric<in TypeAlias>>>
   )
+
+  // Regression test for https://github.com/square/moshi/issues/991
+  @Test fun nullablePrimitiveProperties() {
+    val adapter = moshi.adapter<NullablePrimitives>()
+
+    @Language("JSON")
+    val testOutputJson = """{"oh_boy":"testing","that_is_a_test":3,"test_again":0}"""
+
+    assertThat(adapter.serializeNulls().toJson(NullablePrimitives("testing", 3)))
+        .isEqualTo(testOutputJson)
+
+    @Language("JSON")
+    val testInputJson = """{"oh_boy":"testing","that_is_a_test":3}"""
+    val result = adapter.fromJson(testInputJson)!!
+    assertThat(result.ohboy).isEqualTo("testing")
+    assertThat(result.thatIsATest).isEqualTo(3)
+    assertThat(result.testAgain).isEqualTo(0)
+  }
+
+  // public Testy(String var1, Integer var2, int var3, int var4, DefaultConstructorMarker var5)
```

Comment:
The comment refers to `Testy`, the original class name in the bug report. The test itself uses `NullablePrimitives`.

Reply:
Ah this was accidentally left over, will remove

Reply:
7b4e825